### ISSUE TITLE
Question-writing docs do not describe the `data` dict or data flow

### ISSUE THREAD

**Issue by mwest1066** [PrairieLearn/PrairieLearn#9580]
While the question docs do contain [some basic information](https://prairielearn.readthedocs.io/en/latest/question/#question-serverpy) about the functions in `server.py`, they do not describe the contents of the `data` dict in any detail or how the data-flow occurs in detail. The developer-level element docs [do describe this](https://prairielearn.readthedocs.io/en/latest/devElements/#element-functions) (although see #9579) so we should copy over this information to the question docs and make sure that it's also accurate for `server.py` (I think there might be a couple of tiny differences between elements and questions?).

**Comment by eliotwrobson** [comment]
I think this would be really helpful. The data dictionary and especially the weird control flow is very confusing to new people, and this would go a long way towards making things more accessible.

**Comment by jonatanschroeder** [comment]
I think there should be a separate page for server.py that includes this kind of information.

**Comment by mwest1066** [comment]
I agree that a separate page for `server.py` would be very helpful. The existing question-writing docs page is already too long.

**Comment by reteps** [comment]
This is closed via #11830 - a diagram of data flow is included, as well as a description of the `data` dict.
